taipy/gui/data/pandas_data_accessor.py

```diff
@@ -238,10 +239,12 @@ def __get_data(  # noqa: C901
             nb_rows_max = payload.get("width")
             # view with the requested columns
             if nb_rows_max and nb_rows_max < len(value) / 2:
-                value = value.copy()
-                value["tp_index"] = value.index
-                # we need to be more clever than this :-)
-                value = value.iloc[:: (len(value) // nb_rows_max)]
+                x_column, y_column = columns[1] if len(columns) > 1 else None, columns[0]
+                try:
+                    threshold = payload.get("limitThreshold", 50)
```

Comment:
the 50 value could be defined in the config (give a global default value)

Reply:
You are absolutely right!